packages/react-router-dev/vite/build.ts

```diff
@@ -184,15 +183,18 @@ async function viteBuild(
       optimizeDeps: { force },
       clearScreen,
       logLevel,
-      ...{ __reactRouterEnvironmentBuildContext: environmentBuildContext },
+      ...{
+        __reactRouterPluginContext: ctx,
```

Comment:
We now pass the resolved plugin context object into each build so we can re-use the build manifest rather than re-evaluating the `serverBundles` function whenever our Vite plugin is instantiated.